m2mathew/tmac-website
# Story/60/august 2021 sponsor updates

### Completes [Trello Story](https://trello.com/c/NI74pKtJ/60-august-2021-updates)

--------------
# The issue

August 2021 updates - sponsors

# The solution

- Add back Sponsors page with many more details than before
- Convert lots to TypeScript
- Convert many `class` components to functional components
- Use the `useReducer` hook in both member and sponsor registration forms to help control the data flow (and re-rendering) better
- Test Paypal to make sure it is working
- Make sure the data going to the FireStore database is only what is needed in the synced Google Sheet
- Double check Google Sheets for membership and sponsors.
  - All is well, but the sponsor one seems to duplicate rows that already exist. It can be fixed later ([Trello Card](https://trello.com/c/AkXh2Oi2))
- Update wrong data and links for "By Laws"
- Upgrade a few packges

# Visuals

### Sponsors levels

Class Champion | Gold Medal | Silver Medal
----- | ----- | -----
<img width="805" alt="image" src="https://user-images.githubusercontent.com/11624407/130730857-98729ee6-e79f-4b86-aa6e-0db924988c88.png"> | <img width="755" alt="image" src="https://user-images.githubusercontent.com/11624407/130730877-6f84525b-d64f-41ec-aab8-cf9e7cf33a7c.png"> | <img width="732" alt="image" src="https://user-images.githubusercontent.com/11624407/130730888-b98708c6-9038-46c2-86d6-00d3935aa74e.png">

### Sponsor Registration

Form | Payment invoice | Payment paypal
----- | ----- | -----
<img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731304-4125a9e9-07c0-484d-a2b4-35b23e3789c8.png"> | <img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731318-a3bc9aad-b249-4751-8541-e84ef7799fb6.png"> | <img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731448-f77ff67d-d36e-4b1a-9ab1-1c9f86625c40.png">

Print preview

Invoice | Receipt
----- | -----
<img width="712" alt="image" src="https://user-images.githubusercontent.com/11624407/130731333-b85eb591-5284-4abb-aafc-c4648e0c34bb.png"> | <img width="708" alt="image" src="https://user-images.githubusercontent.com/11624407/130731469-db4a7561-228e-4735-b664-e43ee3331464.png">

### Member Registration

Form | Payment invoice | Payment paypal
----- | ----- | -----
<img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731577-5325ea10-6272-4cfe-910f-8cac3e293e31.png"> | <img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731607-569102b4-f8f9-4f8c-ac94-0f8c00e7796f.png"> | <img width="512" alt="image" src="https://user-images.githubusercontent.com/11624407/130731755-6860a183-8301-4e54-aceb-44861ecbb8c1.png">

Print preview

Invoice | Receipt
----- | -----
<img width="710" alt="image" src="https://user-images.githubusercontent.com/11624407/130731637-9d32b4b0-1d2c-4b3b-97b7-a5e02a7e791f.png"> | <img width="709" alt="image" src="https://user-images.githubusercontent.com/11624407/130731789-3a40683c-e4b4-4c06-b065-027353337d3a.png">
